src/vs/workbench/api/common/extHostTerminalService.ts

```diff
@@ -887,23 +892,30 @@ class EnvironmentVariableCollection implements vscode.EnvironmentVariableCollect
 		this.map = new Map(serialized);
 	}
 
-	get size(): number {
-		return this.map.size;
+	getScopedEnvironmentVariableCollection(scope: vscode.EnvironmentVariableScope | undefined): vscode.EnvironmentVariableCollection {
+		const scopedCollectionKey = this.getScopeKey(scope);
+		let scopedCollection = this.scopedCollections.get(scopedCollectionKey);
+		if (!scopedCollection) {
+			scopedCollection = new ScopedEnvironmentVariableCollection(this, scope);
+			this.scopedCollections.set(scopedCollectionKey, scopedCollection);
+			scopedCollection.onDidChangeCollection(() => this._onDidChangeCollection.fire());
+		}
+		return scopedCollection;
 	}
 
-	replace(variable: string, value: string, scope?: vscode.EnvironmentVariableScope): void {
+	replace(variable: string, value: string, scope: vscode.EnvironmentVariableScope | undefined): void {
```

Comment:
Guessing you just haven't got around to removing the `scope?`?

Reply:
Let me clarify my design, where I have no plans of removing `scope` _internally_.

[`EnvironmentVariableCollection`](https://github.com/microsoft/vscode/pull/182238#discussion_r1192848637) class is used only created once for each extension and then is passed around, that carries information for all the scopes. As `scope` is not present in `constructor`, it is present in each method in the class as a parameter; We should probably rename it to `ExtensionOwnedEnvVarCollection`.

The user exposed `vscode.EnvironmentVariableCollection` is different from this, it does not need `scope` as a parameter.

Reply:
> We should probably rename it to ExtensionOwnedEnvVarCollection

Little confusing as they are by definition owned by an extension

Reply:
We can go something like `GlobalEnvironmentVariableCollection` if it's already clear that they're for a specific extension.

Basically we want to indicate that it is a unified environment variable collection carrying information for all scopes, for a specific extension.

Reply:
`ExtensionOwnedEnvVarCollection` seemed appropriate as it is being used exactly that way in here: https://github.com/microsoft/vscode/blob/9b59b9695bbd90f1be0c45823e506bb42ffd296e/src/vs/platform/terminal/common/environmentVariableCollection.ts#L18

`IExtensionOwnedEnvironmentVariableMutator`(s) came from `ExtensionOwnedEnvVarCollection`.

Reply:
`IExtensionOwnedEnvironmentVariableMutator` is just a mutator object but with the extension id attached. The environment variable collection is extension owned, but the mutator objects are owned by the collection and usually lack this property.

`UnifiedEnvironmentVariableCollection` is my favorite, but I'm not sure we should be renaming it as I believe it's only not-unified in the extension API. Adding "unified" throughout all usages wouldn't add much

Reply:
I like `MergedExtensionEnvVarCollection` because that's what it is, but I agree it probably wouldn't add much. I'll add comments clarifying where it is defined.

Reply:
That's confusing though, merged environment variable collection is already a thing and the merging there means there is only a single value per mutator type + variable.

Reply:
We could remove the "Merge" part, `ExtensionEnvironmentVariableCollection`.